lightclapper/lightclapperclient.py

```diff
@@ -0,0 +1,134 @@
+"""
+lightclapperclient.py
+"""
+import logging
+import time
+import re
+from sqliteDB import LightDB
+from thingspeakreader import ThingSpeakReader
+from constants import (L2_M_5C1_READ_KEY,
+                       L2_M_5C2_READ_KEY,
+                       L2_M_5C1_FEED,
+                       L2_M_5C2_FEED)
+
+POLL_TIME_SECS = 5   # TODO: check this time
+HEADER = 1
+
+
+class LightClapperClient:
+    """
+    Class to parse light clapper data
+    and store in database
+
+    Attributes
+    ----------
+    __reader : ThingSpeakReader
+        Reader of ThingSpeak channel
+    __db : LightClapperDB
+        DB to store LightClapper node data
+
+    Methods
+    -------
+    poll_channel()
+        polls for new data from channel
+    read_from_channel()
+        Read & parse data from channel
+    __add_data_from_channel()
+        Add data from channel if it's not in DB table
+    """
+
+    def __init__(self, key=L2_M_5C1_READ_KEY, feed=L2_M_5C1_FEED):
+        """
+        Initialize light clapper node storage
+
+        Parameters
+        ----------
+        key : str
+            Read API key
+        feed : str
+            Feed number for channel
+        """
+        self.__reader = ThingSpeakReader(key=key, feed=feed)
+
+        with LightDB(db_file='lightclapper.db', name='LightClapper') as db_obj:
+            if not db_obj.table_exists():
+                db_obj.create_table()
+
+    def poll_channel(self):
+        """
+        Poll for new data in channel
+        """
+        try:
+            while True:
+                channel_data = self.read_from_channel()
+                if channel_data:
+                    self.__add_data_from_channel(channel_data)
+                time.sleep(POLL_TIME_SECS)
+        except KeyboardInterrupt:
+            print('Exiting')
+        except BaseException:
+            print('An error or exception occurred!')
+
+    def read_from_channel(self):
+        """
+        Parses data read from channel related to the
+        Light Clapper Node
+
+        Returns
+        -------
+        parsed_data : list
+            data parsed from JSON dict read from channel
+        """
+        read_data = self.__reader.read_from_channel(header=HEADER)
+        feeds = read_data['feeds']
+        parsed_data = []
+        data = {'date': '',
+                'time': '',
+                'lightStatus': ''}
+
+        for f in feeds:
+            date_data = f.get('created_at', '')
+            date_list = re.split('T|Z', date_data)
+
+            if len(date_list) < 2:
+                return None
+
+            data['date'] = date_list[0]
+            data['time'] = date_list[1]
+            data['lightStatus'] = f['field1']
```

Comment:
Should update this later to be f.get('field1', None). Then do a quick check that the value is valid or exists (i.e. not None)